plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Parser.hs

```diff
@@ -31,34 +34,39 @@ import PlutusCore.Parser hiding (parseProgram, parseTerm, program)
 -- Parsers for UPLC terms
 
 -- | A parsable UPLC term.
-type PTerm = UPLC.Term PLC.Name PLC.DefaultUni PLC.DefaultFun SourcePos
+type PTerm = UPLC.Term PLC.Name PLC.DefaultUni PLC.DefaultFun SrcSpan
 
 conTerm :: Parser PTerm
-conTerm = inParens $ UPLC.Constant <$> wordPos "con" <*> constant
+conTerm = lexemeWithSpan $ \sp ->
```

Comment:
This is a case where it may make sense to just call the two versions `withSpan` and `withSpan'`, and make it clear what they do in the Haddock. I think the one that consumes whitespaces is more useful, so I'd call that one `withSpan`.

Reply:
I mean, I don't really have a good, short name for `withSpanAndNoTrailingWhitespace`.

Reply:
Yeah me neither. Fine with giving them non-descriptive names, the "lexeme" bit just feels actively misleading.